Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM haproxy:2.2-alpine
+FROM haproxy:lts-alpine
 
```

Comment:
USER root

Reply:
Running as root is definitely an anti-pattern. I'll see if this works by adding a custom user and ensuring they have access to the mounted docker socket, but in this case it might not be avoidable.

Reply:
In the the 2.2 Docker container the default user is root, in 2.4+ the default user is haproxy, which doesn't have root access inside of the container/image. It's possible that you could solve this in a different way by creating a directory which does not require root access (chown it to haproxy?) in the DockerFile, and then changing haproxy.cfg so that the pidfile gets created there.